tests/test_kat.py

```diff
@@ -30,24 +32,26 @@ def test_sig(sig_name):
         [helpers.path_to_executable('kat_sig'), sig_name],
     )
     output = output.replace("\r\n", "\n")
+    h256 = sha256()
+    h256.update(output.encode())
     kats = []
     avx2_aes_enabled_on_linux = helpers.is_use_option_enabled_by_name('AVX2_INSTRUCTIONS') and helpers.is_use_option_enabled_by_name('AES_INSTRUCTIONS') and platform.system() == 'Linux'
     for filename in os.listdir(os.path.join('tests', 'KATs', 'sig')):
-        if filename.startswith(sig_name + '.') and filename.endswith('.kat'):
+        if filename.startswith(sig_name + '.') and filename.endswith('.sha256'):
             # qtesla's avx2 implementation uses an optimized sampling method that results in different KAT values; we use the correct one (if avx2/aes instructions are available)
             append_kat = False
             if sig_name.startswith('qTesla'):
                 if 'avx2' in filename:
                     if avx2_aes_enabled_on_linux and not helpers.is_build_portable():
                         append_kat = True
                 else:
-                    append_kat = True
+                    if not(avx2_aes_enabled_on_linux and not helpers.is_build_portable()):
+                        append_kat = True
             else:
                 append_kat = True
             if append_kat:
                 with open(os.path.join('tests', 'KATs', 'sig', filename), 'r') as myfile:
-                    kats.append(myfile.read())
-    assert(output in kats)
+                    assert(myfile.read() == h256.hexdigest())
```

Comment:
Similar concerns to the comment above. I also can't make sense of what the `append_kat` stuff is actually doing.

Reply:
`append_kat` was a remnant of the previous logic creating a structure of all KATs (which the one under test needed to match one of). Now streamlined to _one_ perfect match required.